### PR TITLE
Can plot raw fields with new diags

### PR DESCRIPTION
**Pull request by MaxThevenet**
Merge after #824

This is mostly copy-paste of raw fields routines from `FieldsIO.cpp` to `FlushFormatPlotfile.cpp`, with the following differences:
- Interpolation functions are now in namespace `Interpolate`
- Functions WriteCoarseVector and WriteCoarseScalar now take `MultiFab*` instead or `std::unique_ptr<MultiFab>&`
- Some functions, like `WriteRawField`, were renames `WriteRawMF` to avoid linking errors as `WriteRawField` is declared in a .H file.

### CODE REVIEW

**Comment by MaxThevenet** [comment]
Note: I added label WIP because the metadata is incorrect. Will fix this and update the PR.

**Comment by MaxThevenet** [comment]
This should be ready for review.

File: Source/Diagnostics/FlushFormatPlotfile.H [hunk 1]
```diff
@@ -12,13 +12,19 @@ public:
...
+        bool plot_raw_fields_guards,
+        bool plot_rho, bool plot_F) const override final;
 
     /** Write general info of the run into the plotfile */
     void WriteJobInfo(const std::string& dir) const;
     /** Write WarpX-specific plotfile header */
     void WriteWarpXHeader(const std::string& name) const;
-
+    void WriteAllRawFields (const bool plot_raw_fields, const int nlevels,
+                            std::string plotfilename,
```
Comment (atmyers) [inline Source/Diagnostics/FlushFormatPlotfile.H]: `const std::string&`?

Comment (MaxThevenet) [reply Source/Diagnostics/FlushFormatPlotfile.H]: Sure thing, done

**Pull request merged**